src/tls.rs

```diff
@@ -489,24 +498,28 @@ impl Handshake {
 
     pub fn peer_cert(&self) -> Option<Vec<u8>> {
         let peer_cert = unsafe {
-            let mut out: *mut libc::c_uchar = ptr::null_mut();
+            let chain = SSL_get0_peer_certificates(self.as_ptr());
+            if chain.is_null() {
+                return None;
+            }
+            let len = sk_num(chain);
```

Comment:
Don't think this variable is used anywhere other than the check? So can just do:
```rust
if sk_num(chain) <= 0 {
```